pkg/pxc/statefulset.go

```diff
@@ -0,0 +1,315 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	api "github.com/Percona-Lab/percona-xtradb-cluster-operator/pkg/apis/pxc/v1alpha1"
+)
+
+func (h *PXC) newStatefulSetNode(cr *api.PerconaXtraDBCluster) (*appsv1.StatefulSet, error) {
+	ls := map[string]string{
+		"app":       "pxc",
+		"component": cr.Name + "-pxc-nodes",
+		"cluster":   cr.Name,
+	}
+
+	var fsgroup *int64
+	if h.serverVersion.Platform == api.PlatformKubernetes {
+		var tp int64 = 1001
+		fsgroup = &tp
+	}
+
+	resources, err := createResources(cr.Spec.PXC.Resources)
+	if err != nil {
+		return nil, fmt.Errorf("resources: %v", err)
+	}
+
+	rvolStorage, err := resource.ParseQuantity(cr.Spec.PXC.VolumeSpec.Size)
+	if err != nil {
+		return nil, fmt.Errorf("wrong storage resources: %v", err)
+	}
+
+	cfgPV := getConfigVolumes()
+	podObj := corev1.PodSpec{
+		SecurityContext: &corev1.PodSecurityContext{
+			SupplementalGroups: []int64{99},
+			FSGroup:            fsgroup,
+		},
+		Containers: []corev1.Container{{
+			Name:            "node",
+			Image:           cr.Spec.PXC.Image,
+			ImagePullPolicy: corev1.PullAlways,
+			ReadinessProbe: setProbeCmd(&corev1.Probe{
+				InitialDelaySeconds: 15,
+				TimeoutSeconds:      15,
+				PeriodSeconds:       30,
+				FailureThreshold:    5,
+			}, "/usr/bin/clustercheck.sh"),
+			LivenessProbe: setProbeCmd(&corev1.Probe{
+				InitialDelaySeconds: 300,
+				TimeoutSeconds:      5,
+				PeriodSeconds:       10,
+			}, "/usr/bin/clustercheck.sh"),
+			Ports: []corev1.ContainerPort{
+				{
+					ContainerPort: 3306,
+					Name:          "mysql",
+				},
+			},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					Name:      "datadir",
+					MountPath: "/var/lib/mysql",
+				},
+				{
+					Name:      "config-volume",
+					MountPath: "/etc/mysql/conf.d/",
+				},
+			},
+			Env: []corev1.EnvVar{
+				{
+					Name: "MYSQL_ROOT_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: secretKeySelector(cr.Spec.SecretsName, "root"),
+					},
+				},
+				{
+					Name: "CLUSTERCHECK_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: secretKeySelector(cr.Spec.SecretsName, "clustercheck"),
+					},
+				},
+				{
+					Name: "XTRABACKUP_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: secretKeySelector(cr.Spec.SecretsName, "xtrabackup"),
+					},
+				},
+				{
+					Name: "MONITOR_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: secretKeySelector(cr.Spec.SecretsName, "monitor"),
+					},
+				},
+				{
+					Name: "CLUSTERCHECK_PASSWORD",
+					ValueFrom: &corev1.EnvVarSource{
+						SecretKeyRef: secretKeySelector(cr.Spec.SecretsName, "clustercheck"),
+					},
+				},
+			},
+			Resources: resources,
+		}},
+		Volumes: []corev1.Volume{
+			cfgPV,
+		},
+	}
+
+	pvcObj := corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "datadir",
+		},
+		Spec: corev1.PersistentVolumeClaimSpec{
```

Comment:
hmm, should we specify persistentVolumeReclaimPolicy: Delete 
what do you think?

Reply:
does it make sense to allow user to define the reclaim policy config with `Delete` defaults? or just make it `Delete`?

Reply:
Ok, it's not an option of PV, not PVC. And this is inherites from the Storage Class. So there is no way to set it in a Stateful Set. The only way is to get created PVs and redefine this options by updating. But it looks redundant and weird. Way better just use appropriate Storage Class.